nist/int.go

```diff
@@ -140,7 +140,9 @@ func (i *Int) Set(a abstract.Scalar) abstract.Scalar {
 }
 
 func (i *Int) Clone() abstract.Scalar {
-	return &Int{V: i.V, M: i.M, BO: i.BO}
+	v := new(big.Int).Set(&i.V)
+	m := new(big.Int).Set(i.M)
+	return &Int{V: *v, M: m, BO: i.BO}
```

Comment:
So the complete is:
```
	ni := new(Int).Init(&i.V, i.M)
	ni.BO = i.BO
	return ni
```

So either we take your solution, or we re-write `Int.Init` to take `BO` as a parameter, which might makes some sense, given that the `V` -argument to `Init` could be in the wrong endianness. But I don't know if `BO` is really relevant in `i.V.Set(V).Mod(&i.V, M)` - probably not.

Your call, as Ismail is not here to vote ;)